interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -210,44 +224,62 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
       case e: OpEQ => checkEqualityLikeOperation(validationContext, e, current, isEquality = true)
       case e: OpNE => checkEqualityLikeOperation(validationContext, e, current, isEquality = false)
 
-      case e: OpAnd => withChildrenOfType[Boolean](TypingResultWithContext(Typed[Boolean]))
-      case e: OpOr => withChildrenOfType[Boolean](TypingResultWithContext(Typed[Boolean]))
-      case e: OpGE => withChildrenOfType[Number](TypingResultWithContext(Typed[Boolean]))
-      case e: OpGT => withChildrenOfType[Number](TypingResultWithContext(Typed[Boolean]))
-      case e: OpLE => withChildrenOfType[Number](TypingResultWithContext(Typed[Boolean]))
-      case e: OpLT => withChildrenOfType[Number](TypingResultWithContext(Typed[Boolean]))
+      case e: OpAnd => withTwoChildrenOfType[Boolean, Boolean](_ && _)
+      case e: OpOr => withTwoChildrenOfType[Boolean, Boolean](_ || _)
+      case e: OpGE => withTwoChildrenOfType[Number, Boolean](MathUtils.greaterOrEqual)
```

Comment:
Number and Boolean can't be inferred from MathUtils.greaterOrEqual signature by scala? :thinking: